client/state/data-layer/wpcom/comments/index.js

```diff
@@ -156,6 +160,35 @@ export const announceFailure = ( { dispatch, getState }, { siteId, postId } ) =>
 	dispatch( errorNotice( error ) );
 };
 
+export const deleteComment = ( { dispatch }, action ) => {
+	const { siteId, commentId } = action;
+	dispatch(
+		http( {
+			method: 'POST',
+			apiVersion: '1.1',
+			path: `/sites/${ siteId }/comments/${ commentId }/delete`,
+			onSuccess: {
+				...action,
+				type: COMMENTS_DELETE_SUCCESS,
+			},
+			onFailure: {
+				...action,
+				type: COMMENTS_DELETE_FAILURE,
+			},
+		} )
+	);
+};
+
+export const removeCommentFromState = ( { dispatch }, action ) => {
+	const { siteId, postId, commentId } = action;
+	dispatch( removeComment( siteId, postId, commentId ) );
```

Comment:
actually here I don't think we need to do anything, right? if the comment deletion was successful then we're done because the reducer should have removed the comment when it saw `COMMENTS_DELETE`. Thoughts?